src/bin/mdbook-xgettext.rs

```diff
@@ -222,4 +497,172 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn test_create_catalog_nested_directories() -> anyhow::Result<()> {
+        let (ctx, _tmp) = create_render_context(&[
+            ("book.toml", "[book]"),
+            (
+                "src/SUMMARY.md",
+                "- [The Foo Chapter](foo.md)\n\
+                \t- [The Bar Section](foo/bar.md)\n\
+                \t\t- [The Baz Subsection](foo/bar/baz.md)",
+            ),
+            (
+                "src/foo.md",
+                "# How to Foo\n\
+                 \n\
+                 The first paragraph about Foo.\n",
+            ),
+            (
+                "src/foo/bar.md",
+                "# How to Bar\n\
+                 \n\
+                 The first paragraph about Bar.\n",
+            ),
+            (
+                "src/foo/bar/baz.md",
+                "# How to Baz\n\
+                 \n\
+                 The first paragraph about Baz.\n",
+            ),
+        ])?;
+
+        // Using a depth of 0 to include all messages in a single template file.
+        let catalogs = create_catalogs(&ctx, 0, "messages.pot")?;
+        let catalog = catalogs.get("messages.pot").unwrap();
+
+        for msg in catalog.messages() {
+            assert!(!msg.is_translated());
+        }
+
+        assert_eq!(
+            catalog
+                .messages()
+                .map(|msg| (msg.source(), msg.msgid()))
+                .collect::<Vec<(&str, &str)>>(),
+            &[
+                ("src/SUMMARY.md:1", "The Foo Chapter"),
+                ("src/SUMMARY.md:2", "The Bar Section"),
+                ("src/SUMMARY.md:3", "The Baz Subsection"),
+                ("src/foo.md:1", "How to Foo"),
+                ("src/foo.md:3", "The first paragraph about Foo."),
+                ("src/foo/bar.md:1", "How to Bar"),
+                ("src/foo/bar.md:3", "The first paragraph about Bar."),
+                ("src/foo/bar/baz.md:1", "How to Baz"),
+                ("src/foo/bar/baz.md:3", "The first paragraph about Baz."),
+            ]
+        );
+
+        Ok(())
+    }
+
+    // Helper constants written as functions.
```

Comment:
I'm not entirely sure what this is doing — I have a strong feeling that we don't need to care about `const` here?

Reply:
Removed this comment - my reason for including it was to explain why the 'helper' variables were extracted into functions since they can't directly be written as variables